src/runtime/methodbinder.cs

```diff
@@ -378,6 +378,7 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw, MethodBase info, Meth
                 var outs = 0;
                 var margs = TryConvertArguments(pi, paramsArray, args, pynargs, kwargDict, defaultArgList,
                     needsResolution: _methods.Length > 1,  // If there's more than one possible match.
+                    setError: _methods.Length == 1 && !isGeneric,   // True when there are no alternatives.
```

Comment:
This is problematic in the following scenario:
```python
try:
  UInt64Method(some_val)
except OverflowError:
  print('value is too large')
```

When somebody adds `UInt64Method(string)`, the code above will stop working despite the new overload not getting matched.

Reply:
An option is to set `__cause__` of the "unable to find overload" `TypeError`  to an instance of `AggregateException`

Reply:
Yes, there will be an inconsistency in error messages depending on whether a method is overloaded.  I think this is the best that can be done in the current bad situation wrt overloading.

Reply:
The situation is mostly bad due to unexpected default conversions, which will be gone in 3.0.0. We should not introduce an inconsistent behavior to patch it temporarily, when it is possible to improve it without inconsistencies.

Reply:
I think the solution you want is to preserve "No method matches" as the top-level error, but include the OverflowError as an inner error.  I will change that.

Reply:
Right, I could have stated it clearly.

Consider also collecting errors from all overloads.